plugins/package-managers/cocoapods/src/main/kotlin/Lockfile.kt

```diff
@@ -95,16 +97,16 @@ internal data class Lockfile(
     )
 }
 
-internal fun String.parseLockfile(): Lockfile {
-    val root = Yaml.default.parseToYamlNode(this).yamlMap
+internal fun File.parseLockfile(): Lockfile {
+    val root = Yaml.default.parseToYamlNode(this.readText()).yamlMap
 
     val externalSources = root.get<YamlMap>("EXTERNAL SOURCES")?.entries.orEmpty().map {
         val name = it.key.content
         val node = it.value.yamlMap
 
         val externalSource = ExternalSource(
-            path = node.get<YamlScalar>(":path")?.content,
-            podspec = node.get<YamlScalar>(":podspec")?.content
+            path = node.get<YamlScalar>(":path")?.content?.let { rawPath -> parentFile.resolve(rawPath).path },
+            podspec = node.get<YamlScalar>(":podspec")?.content?.let { rawPath -> parentFile.resolve(rawPath).path }
```

Comment:
I believe the new functionality goes beyond just parsing the lockfile and therefore should not be part of this file.
Can path resolution be done outside of this file, e.g. in CocoaPods?

Reply:
The problem is that the `path` and `podspec` properties are used by the `PodDependencyHandler`. I cannot add a reference to the lock file to the handler because it is build during the initialization of the `CocoaPods` class (I guess because the handler maintains a cache of the `Podspec`s).
Additionally, I cannot pass the lock file to any of the functions of the handler because they are fixed by the interface `DependencyHandler`.

Therefore, the only solution I found was to put the resolution in the `Lockfile` class. If you have any other idea...

Reply:
I see. Would it be an option to do `fun Podspec.resolvedRelativePaths(base: File): Podspec`?

This way we keep that majority of the logic file agnostic, so that you can run it given any string, as this can be convenient.

Reply:
As solution, maybe, would be in `CocoaPods`, when `lockfileData.dependencies` is accessed, to do a deep copy with the resolved paths.
Would it be better ? I have some concerns duplicating a potentially big data structure just for that.

Reply:
But the handler does not know anything about the lockfile nor its path. So I wouldn't know which value to pass to this `base` parameter.

Reply:
I see. (I misunderstood your above comment and assumed the handler would be the one calling the parse function). However, would it work if the function was in `CocoaPods.kt`, as the only caller seems to be here ? https://github.com/oss-review-toolkit/ort/blob/737038d25ed9fa426e6f27e98fa14b80b1180036/plugins/package-managers/cocoapods/src/main/kotlin/CocoaPods.kt#L132

Reply:
So to be clear, you suggest something like `lockfile.readText().parseLockfile().resolveRelativePaths(lockfile)` were you postprocess the parsed `Lockfile` to resolve all the paths, at the cost of copying the `Lockfile` object ?

Or maybe more elegant:
`val dependencies = lockfileData.dependencies.mapNotNull { it.resolvedPod.withResolvesRelativePaths(lockfile) }`

Reply:
@fviernau  Please have a look. I think I fixed it.